backend/backend-standard-and-best-practices.md

```diff
@@ -0,0 +1,278 @@
+- Depending on the case and the receiver of the response, it may not be necessary to send both the `code` and `message` parameters.
+
+## 4- API Versioning
+
+We decided to implement API versioning via the use of headers.
+
+
+- Java example:
+```java
+/*
+http://localhost:8080/person/header
+headers=[X-API-VERSION=1]
+http://localhost:8080/person/header
+headers=[X-API-VERSION=2]
+Implementations are shown below:
+*/
+
+  @GetMapping(value = "/student/header", headers = "X-API-VERSION=1")
+  public StudentV1 headerV1() {
+    return new StudentV1("Bob Charlie");
+  }
+
+  @GetMapping(value = "/student/header", headers = "X-API-VERSION=2")
+  public StudentV2 headerV2() {
+    return new StudentV2(new Name("Bob", "Charlie"));
+  }
+
+```
+
+- Rails example (using [Versionist])
+```ruby
+MyApi::Application.routes.draw do
+  api_version(:module => "V20120317", header: {name: "X-API-VERSION", value: "v20120317"}) do
+    resources :bars
+  end
+end
+```
+
+- Node/Express example (using [Express Routes Versioning])
+```javascript
+var app = require('express')();
+var versionRoutes = require('express-routes-versioning')();
+app.listen(3000);
+app.use(function(req, res, next) {
+    //req.version is used to determine the version
+   req.version = req.headers['X-API-VERSION'];
+   next();
+});
+app.get('/users', versionRoutes({
+   "1.0.0": respondV1,
+   "~2.2.1": respondV2
+}));
+
+// curl -s -H 'accept-version: 1.0.0' localhost:3000/users
+// version 1.0.0 or 1.0 or 1 !
+function respondV1(req, res, next) {
+   res.status(200).send('ok v1');
+}
+
+//curl -s -H 'accept-version: 2.2.0' localhost:3000/users
+//Anything from 2.2.0 to 2.2.9
+function respondV2(req, res, next) {
+   res.status(200).send('ok v2');
+}
+```
+
+### 5- Response Status Codes
+
+There are many _status codes_ to use in request responses. \
+Most commonly used are:
+
+* **200 OK**: Basic successful response. Depends on currently used HTTP method.
+* **201 CREATED**: Successful response meaning a new resource has been created. Most commonly used with POST and sometimes PUT.
+* **204 NO CONTENT**: Successful response without content in body.
+* **400 BAD REQUEST**: Request was not formatted correctly and the server cannot interpret it (syntax error).
+* **401 UNAUTHORIZED**: The client must authenticate itself to get the requested response.
+* **403 FORBIDDEN**: Incorrect level of authorization to use a specific resource.
+* **404 NOT FOUND**: Specified resource was not found.
+* **422 UNPROCESSABLE ENTITY**: Must be used when the server cannot handle the request as is. For example, a parameter image cannot be read correctly or some required parameters are missing.
```

Comment:
`Must be used when the server cannot handle the request as it is`, I think there is an `it` word missing